packages/core/src/agent/MessageSender.ts

```diff
@@ -160,16 +160,24 @@ export class MessageSender {
       transportPriority?: TransportPriorityOptions
     }
   ) {
-    const { connection, payload } = outboundMessage
+    const { connection, payload, sessionId } = outboundMessage
     const errors: Error[] = []
 
     this.logger.debug('Send outbound message', {
       message: payload,
       connectionId: connection.id,
     })
 
+    let session: TransportSession | undefined
+
+    if (sessionId) {
+      session = this.transportService.findSessionById(sessionId)
+    }
+    if (!session) {
+      session = this.transportService.findSessionByConnectionId(connection.id)
```

Comment:
I tried that, but that won't work in this case because I want to find as session by id if the session id is present, and if there is no session form that first step then I want to find the session by connection id. 

So it could be that both methods are called, and with a ternerary only one of the two will be called